course_info/static/course_info/js/controllers/IndexController.js

```diff
@@ -124,9 +142,55 @@
 
         $scope.initializeDatatable = function() {
             $scope.dataTable = $('#courseInfoDT').DataTable({
-                data: $scope.courseInfo,
-                dataSrc: '',  // data is an array
-                // todo: p_r_ocessing?...
+                serverSide: true,
+                deferLoading: true,
+                ajax: function(data, callback, settings) {
+                    $scope.$apply(function(){
+                        $scope.searchInProgress = true;
+                    });
+                    var queryParameters = {};
+                    if ($scope.queryString.trim() != '') {
+                        queryParameters.search = $scope.queryString.trim();
+                    }
+                    if ($scope.filtersApplied) {
+                        for (var key in $scope.filters) {
+                            var f = $scope.filters[key];
+                            if (f.query) { queryParameters[f.key] = f.value; }
+                        }
+                    }
+                    queryParameters.offset = data.start;
+                    queryParameters.limit = data.length;
+                    var order = data.order[0];
+                    queryParameters.ordering = (order.dir == 'desc' ? '-' : '') + $scope.columnFieldMap[order.column];
+                    $.ajax({
+                        url: '/icommons_rest_api/api/course/v2/course_instances',
+                        method: 'GET',
+                        data: queryParameters,
+                        dataType: 'json',
+                        success: function(data, textStatus, jqXHR) {
+                            var results = data.results;
+                            var resultsLength = data.results.length;
+                            var processedData = [];
+                            for (var i = 0; i < resultsLength; i++) {
+                                processedData.push($scope.courseInstanceToTable(results[i]));
+                            }
+                            callback({
+                                recordsTotal: data.count,
+                                recordsFiltered: data.count,
```

Comment:
I guess this means that recordsFiltered/recordsTotal distinction doesn't make much sense in our context, since we're not filtering within a larger dataset, but we have to return this as part of datatables I suppose? A comment in here about how our API treats these might be nice, since a useful recordsTotal won't be returned by the API it seems.